docs/src/site/twiki/UserExtensions.twiki

```diff
@@ -0,0 +1,232 @@
+    |-- META
+        |-- properties file // This file will be ‘cat’ when extension user issues the “definition” command.
+    |-- libs // Libraries to used during build time (instantiation of extension) and runtime
+        |-- build
+    |-- resources (OPTIONAL)// Other resource files to be used during build time (instantiation of extension)
+        |-- build
+</verbatim>
+The jar supplied in the build directory should be packaged with the following file META-INF/services/org.apache.falcon.extensions.ExtensionBuilder
+containing the full class name of the class which implements ExtensionBuilder.
+
+
+
+---++ Extension User
+The Extension user is the one who will use and instantiate the extensions that are available.
+
+---+++Enumerate Extensions(including trusted extensions)
+---++++Command: falcon extension -enumerate
+Sample Output (Could be tabular format too, instead of plain JSON): {
+<verbatim>
+  "extensions": [
+    {
+      "name": "hdfs-mirroring",
+      "type": "Trusted extension",
+      "description": "This extension implements replicating arbitrary directories on HDFS from one Hadoop cluster to another Hadoop cluster. This piggy backs on replication solution in Falcon which uses the DistCp tool.",
+      “location”: “fs://<falcon-home>/extensions/hdfs-mirroring”
+    },
+    {
+      "name": "data-quality-check",
+      "type": "User extension",
+      "description": "This extension allows you to check if the data complies with the schema."
+      “location” : “hdfs://<nn:port>/projects/<user-dir>/falcon-extensions/…”
+    },
+    "totalResults": 2
+]
+</verbatim>
+---+++ Get description/details of an extension
+Displays the README supplied by the extension Developer “as-it-is”.
+---++++ Command: falcon extension -describe -extensionName <extension name>
+<verbatim>
+Sample Command : falcon extension -describe -extensionName data-quality-check
+Sample Output :
+The data quality extension allows you…..
+
+</verbatim>
+
+---+++Details of an extension
+Gives the details that Extension Developer has supplied during extension registration.
+---++++Command: falcon extension -detail -extensionName <extension name>
+<verbatim>
+Sample Command : falcon extension -detail -extensionName hdfs-mirroring
+Sample Output :
+{
+      "name": "hdfs-mirroring",
+      "type": "Trusted extension",
+      "description": "This extension implements replicating arbitrary directories on HDFS from one Hadoop cluster to another Hadoop cluster. This piggy backs on replication solution in Falcon which uses the DistCp tool.",
+      “location”: “fs://<falcon-home>/extensions/hdfs-mirroring”
+}
+</verbatim>
+
+---+++ Extension Instatiation validation.
+Validates the configs supplied for an instantiation of the given extension.
+<verbatim>
+Sample Command: falcon extension -validate -extensionName data-quality-check -file extensions/hdfs-mirroring//META/hdfs-mirroring-properties.json
```

Comment:
jobName is required for validate feature.
https://github.com/apache/falcon/blob/master/cli/src/main/java/org/apache/falcon/cli/FalconExtensionCLI.java#L134